### ISSUE TITLE
Cannot build without cython and no error given, when compiling build_rir

### ISSUE THREAD

**Issue by oleg-alexandrov** [LCAV/pyroomacoustics#163]
Greetings. Thank you for the very cute pyroomacoustics package. I found it very useful. 

I found one issue when building from source that I found baffling. If I do, with a latest python (version 3.7)

python setup.py build_ext 

in the pyroomacoustics freshly cloned directory, I get an error about build_rir.c missing. 

The reason, is after some digging, that cython was missing, so build_rir.c was not created. I guess though that setup.py should check for that, and fail with a clear error message, rather than allowing things to get to this point, as things are quite confusing, with some c files being present, but build_rir.c missing (I thought it was not checked in or something).

A minor thing, of course, but may help with usability. Thanks.

**Comment by fakufaku** [comment]
Hi @oleg-alexandrov , thanks for reporting this! We have actually run into this in the past (see #117). The issue is due to an idiosyncrasy of pip that doesn't necessarily install the dependencies prior to running the setup file. I recall looking things up and that there was no good solution for this. The issue happened on linux. On mac and windows, I think pip was smart enough to get dependencies first...
However, I also found out that modern setuptools/distutils should be smart enough to detect files to cythonize and I tried to fix the problem in PR #122.

So can you please let me know which OS and version of pyroomacoustics you are using ?

Also, if you know of a solution for this problem, I'd be also very interested to hear it.

**Comment by oleg-alexandrov** [comment]
This looks nasty. I detected this on Ubuntu 16.04, which is getting old. 

I don't know much about how to deal with cython. I made a pull request (https://github.com/LCAV/pyroomacoustics/pull/164) to at least improve the error message. Given that this occurs only on older (and soon unsupported) systems, maybe this is good enough.

**Comment by fakufaku** [comment]
Thanks! It actually seems that [PEP 518](https://www.python.org/dev/peps/pep-0518/) provides a solution via an extra project file. I have tried on a linux virtual machine and with the project file it first downloads the dependencies.

**Comment by oleg-alexandrov** [comment]
Thanks. I tried your fix (commit 4681b3c) but it did not work on my Ubuntu 16 machine with python 2.7. Oh well. Old machine and old python.  At least the error message provides a hint now. 

For the record, python 2.7 is causing pain even when using pip to install pyroomacoustics, rather than build it from scratch as i tried. Pip has to be told specifically what versions of packages to install, as pip (at least my old pip) always aims for newest versions of packages and those work only with Python 3. For the record, here's a combination that worked for me: 

pip install numpy==1.15.4 scipy==0.18 pillow==6 matplotlib==2.0.0  pyroomacoustics==0.3.1 

I know all developers want to forget about python 2.7. The problem is, for example, that ROS is still stuck with it, and I use pyroomacoustics as a ROS node to simulate a robot's microphone. 

Sorry for the long rant. :)

**Comment by fakufaku** [comment]
Ah, python 2.7... :expressionless: It is starting to cause the continuous integration to break and I was thinking of dropping support altogether. I understand your frustration, and you make a good case, but we are also stretched thin.

Anyway, is it possible to use anaconda (or rather [miniconda](https://docs.conda.io/en/latest/miniconda.html)) on ROS ? I just tested on (recent) Ubuntu virtual machine and the installation was smooth.

Assuming miniconda installed, I did the following
```
conda create -n prapy27 python=2.7
conda activate prapy27
python -m pip install -e https://github.com/LCAV/pyroomacoustics
```
and to the master branch installed without problem.
